includes/connect/class-wc-stripe-connect.php

```diff
@@ -205,17 +183,44 @@ private function get_default_stripe_config() {
 			return $result;
 		}
 
-		public function is_connected() {
-
+		/**
+		 * Determines if the store is connected to Stripe.
+		 *
+		 * @param string $mode Optional. The mode to check. 'live' or 'test' - if not provided, the currently enabled mode will be checked.
+		 * @return bool True if connected, false otherwise.
+		 */
+		public function is_connected( $mode = null ) {
 			$options = get_option( self::SETTINGS_OPTION, [] );
 
-			if ( isset( $options['testmode'] ) && 'yes' === $options['testmode'] ) {
+			// If the mode is not provided, we'll check the current mode.
+			if ( is_null( $mode ) ) {
+				$mode = isset( $options['testmode'] ) && 'yes' === $options['testmode'] ? 'test' : 'live';
+			}
+
+			if ( 'test' === $mode ) {
 				return isset( $options['test_publishable_key'], $options['test_secret_key'] ) && trim( $options['test_publishable_key'] ) && trim( $options['test_secret_key'] );
 			} else {
 				return isset( $options['publishable_key'], $options['secret_key'] ) && trim( $options['publishable_key'] ) && trim( $options['secret_key'] );
 			}
 		}
 
+		/**
+		 * Determines if the store is connected to Stripe via OAuth.
+		 *
+		 * @param string $mode Optional. The mode to check. 'live' or 'test' (default: 'live').
+		 * @return bool True if connected via OAuth, false otherwise.
+		 */
+		public function is_connected_via_oauth( $mode = 'live' ) {
+			if ( ! $this->is_connected( $mode ) ) {
+				return false;
+			}
+
+			$options = get_option( self::SETTINGS_OPTION, [] );
+			$key     = 'test' === $mode ? 'test_connection_type' : 'connection_type';
+
+			return isset( $options[ $key ] ) && in_array( $options[ $key ], [ 'connect' ], true );
```

Comment:
I've used `in_array()` here even though there's only a single `connect` type. The reason  for that is that there will eventually be a `app` type too. @diegocurbelo does that make sense to you or should I just leave this limited to `connect` and `app` will be handled later.

Reply:
I like this implementation, both are going to be handled the same one.